fdb-record-layer-spatial/src/main/java/com/apple/foundationdb/record/spatial/geophile/GeophileSpatial.java

```diff
@@ -0,0 +1,254 @@
+/*
+ * GeophileSpatial.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2019 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.spatial.geophile;
+
+import com.apple.foundationdb.annotation.API;
+import com.geophile.z.Space;
+import com.geophile.z.SpatialObject;
+import com.geophile.z.spatialobject.jts.JTS;
+import com.geophile.z.spatialobject.jts.JTSSpatialObject;
+import org.locationtech.jts.geom.CoordinateSequence;
+import org.locationtech.jts.geom.CoordinateSequenceFilter;
+import org.locationtech.jts.geom.Geometry;
+import org.locationtech.jts.geom.GeometryFactory;
+import org.locationtech.jts.geom.Point;
+import org.locationtech.jts.io.ParseException;
+import org.locationtech.jts.io.WKBReader;
+import org.locationtech.jts.io.WKBWriter;
+import org.locationtech.jts.io.WKTReader;
+import org.locationtech.jts.io.WKTWriter;
+import org.locationtech.jts.io.geojson.GeoJsonReader;
+import org.locationtech.jts.io.geojson.GeoJsonWriter;
+import org.slf4j.bridge.SLF4JBridgeHandler;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Helper methods for interfacing with Geophile, specifically for implementing geospatial with lat, lon coordinates.
+ */
+@API(API.Status.EXPERIMENTAL)
+@SuppressWarnings("checkstyle:abbreviationaswordinname")    // Allow WKB, WKT
+class GeophileSpatial {
+    /*
+      The lat/lon coordinate system is
+
+      - latitude: -90.0 to +90.0
+      - longitude: -180.0 to 180.0 with wraparound
+
+      The interleave pattern is [lon, lat, lon, lat, ..., lon], reflecting the fact that longitude covers a numeric range
+      twice that of latitude.
```

Comment:
I figured out what this meant by looking at Geophile docs, but I wonder if it's worth adding a few more lines of justification here to help a non-expert get up to speed faster on why we chose this interleave pattern.

I think I deleted all the quizzical comments related to this but if I missed one you can ignore it. :)

Reply:
I expanded the comment.